code/modules/wod13/onyxcombat.dm

```diff
@@ -578,3 +551,83 @@
 				elysium_checks = 0
 				if(HAS_TRAIT(src, TRAIT_ELYSIUM))
 					REMOVE_TRAIT(src, TRAIT_ELYSIUM, "elysium")
+
+/mob/living/carbon/human
+	var/current_police_raid = FALSE //lock to prevent multiple handle_police_raid() procs from going at the same time
+
+/mob/living/carbon/human/proc/handle_police_raid() //hooo boy
+	if(warrant)
+		if(key) //are we a player?
+			if(stat != DEAD) //are we Not Dead?
+				if(istype(get_area(src), /area/vtm))
+					var/area/vtm/V = get_area(src)
+					if(V.upper && !current_police_raid && last_raid+next_raid < world.time)
+						current_police_raid = TRUE
+						var/warning_msg = pick("You see them searching above...", "The rumble of helicopter blades draws closer...")
+						to_chat(src, "<span class='warning'>[warning_msg]</span>")
+						playsound(loc, 'code/modules/wod13/sounds/helicopter_far.ogg', 50, FALSE)
+						sleep(60)
+						warning_msg = pick("You really need to get out of sight now.", "They'll spot you any moment now.")
+						to_chat(src, "<span class='warning'>[warning_msg]</span>")
+						sleep(40)
+						V = get_area(src)
+						if(!istype(get_area(src), /area/vtm) || !V.upper) //did the player hide in time?
+							to_chat(src, "<span class='warning'>You slip back into the shadows before they notice.</span>")
+							playsound(loc, 'code/modules/wod13/sounds/helicopter_leaving.ogg', 45, FALSE)
+							last_nonraid = world.time
+						else
+							if(stat != DEAD)
+								last_showed = world.time
+								var/loclist = view(3,src) - view(2, src) //spawn enemies in a ring with dist = 3 from player in visible tile
+
+								sleep(20)
```

Comment:
Same issue for these. Asynchronous calls will make sure the life() function stops for nothing